src/highlight.c

```diff
@@ -309,8 +309,8 @@ static char *(highlight_init_dark[]) = {
     CENT("SignColumn term=standout ctermbg=DarkGrey ctermfg=Cyan",
 	 "SignColumn term=standout ctermbg=DarkGrey ctermfg=Cyan guibg=Grey guifg=Cyan"),
 #endif
-    CENT("Visual term=reverse",
-	 "Visual term=reverse guibg=DarkGrey"),
+    CENT("Visual ctermbg=235 ctermfg=LightGrey",
+	 "Visual ctermbg=235 ctermfg=LightGrey guifg=LightGrey guibg=#262626"),
```

Comment:
Probably a bit lighter guibg (#444444) would be better:
![image](https://github.com/vim/vim/assets/234774/055ae628-b1c4-4479-bb11-758a01ee5929)

Reply:
I think `ctermbg=235` is too optimistic, I think DarkGray would be a safer bet

with `ctermfg=White`:
![image](https://github.com/vim/vim/assets/234774/d7aa683c-ba0b-4a4e-bc61-0c6ba24f88e9)

with `ctermfg=Black`:
![image](https://github.com/vim/vim/assets/234774/617d7793-53b9-4534-bbf7-23ef4b89c930)

@romainl, @neutaaaaan what do you think?

Reply:
I think your tweaks to the 24bit version are fine. It's a completely different colourscheme anyway.

On the other hand, I disagree with the changes made to the 256c selections , as both versions are *inherently* 8c colourschemes that haphazardly use colors from the extended palette. 
Being the default colourscheme, this is actually a fairly sensible behavior to ensure the best compatibility possible regardless of configuration mishaps, and I've extended that behavior by bolting as many of the hl groups used during common editing down in my back of the napkin patch referenced earlier.

This is what the end user will see first in most cases.
This is what the end user will end up using in most cases.
We cannot ever assume their environment will be set up correctly.
We cannot ever assume vim will be able to autodetect the terminal background color.
The only safe assumption, is that the light version of the colorscheme is going to be loaded, *eventually*, and rework that as the new background agnostic default.

Reply:
Grey is hard to get right in 8c and 16c.

Assuming the user's terminal emulator's palette follows the same structure as the default xterm palette, we only have *two* greys that can be reliably used for background in 16c:

- 7 is very light at `#dfdfdf`,
- 8 is very dark at `#6c6c6c`.

This limits what can be done for visual selection if we insist on using grey:

- 0 on 7 works,
- 7 on 8 works,
- 15 on 8 works.

In 8c, the only viable grey background is 7, which restricts us to:

- 0 on 7

My advice would thus be to use `ctermbg=grey ctermfg=black` for both 8c (because of 8c limitations) and 16c (because of consistency with 8c):

<img width="697" alt="Capture d’écran 2024-01-23 à 17 41 56" src="https://github.com/vim/vim/assets/344335/9ad9411d-8e74-4c08-ab33-636998145b73">

Note that the limited grey palette is quite problematic when it comes to UI. A mostly grey chrome is quite practical because it doesn't distract from the syntax highlighting. The problem is that we have lots of greys to play with in GUI and 256c but very few in 16c and 8c. If we shoot for consistency, we should, IMO keep all the greys we can for the chrome (status line, tab line, completion menu, etc.). This means using colors as much as possible for in-buffer highlighting.

Something like the following might be more effective?

<img width="567" alt="Capture d’écran 2024-01-23 à 18 03 00" src="https://github.com/vim/vim/assets/344335/e5c0b908-4716-4690-964d-97976f8e4bf9">

Reply:
> Grey is hard to get right in 8c and 16c.
Which is why I don't use it.

The only available colors that in the default colourschemes don't outright disappear inside diffs would be green and yellow.
I settled on black foreground, yellow background long ago.

Please take a look at [this branch](https://github.com/neutaaaaan/vim/tree/light_colourscheme_fixes)

Reply:
I'll revert the gui changes, even so I think it doesn't work very well with 'termguicolors' set in dark mode:
![grafik](https://github.com/vim/vim/assets/244927/453adaea-c829-4e31-95bb-07f599f43cd7)

I'll also make the suggested changes to use `ctermbg=grey ctermfg=black` then.